src/utilities/findDescriptionChanges.js

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-restricted-syntax */
+// @flow
+
+import {
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+} from '../type/definition';
+import type { GraphQLFieldMap } from '../type/definition';
+import { GraphQLSchema } from '../type/schema';
+
+/**
+ * Given two schemas, returns an Array containing descriptions of any
+ * descriptions that are new or changed and need review.
+ */
+export function findDescriptionChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema,
+): Array<string> {
+  const oldTypeMap = oldSchema.getTypeMap();
+  const newTypeMap = newSchema.getTypeMap();
+
+  const descriptionChanges: Array<string> = [];
+
+  Object.keys(newTypeMap).forEach(typeName => {
+    const oldType = oldTypeMap[typeName];
+    const newType = newTypeMap[typeName];
+
+    if (newType.description) {
+      if (!oldType) {
+        descriptionChanges.push(
+          `Description added on new type ${newType.name}.`,
+        );
+      } else if (!oldType.description) {
+        descriptionChanges.push(`Description added on type ${newType.name}.`);
+      } else if (oldType.description !== newType.description) {
+        descriptionChanges.push(`Description changed on type ${newType.name}.`);
+      }
+    }
+
+    if (oldType && !(newType instanceof oldType.constructor)) {
+      return;
+    }
+
+    if (
+      newType instanceof GraphQLObjectType ||
+      newType instanceof GraphQLInterfaceType ||
+      newType instanceof GraphQLInputObjectType
+    ) {
+      const oldTypeFields: ?GraphQLFieldMap<*, *> = oldType
```

Comment:
Should this differentiate between the old type not existing vs the old type existing but the field not existing?

It could be overwhelming to see a ton of "Description added on new field" for every field on a type that was newly added, that might not be helpful.

Instead perhaps "New type added with description" and "New field added with description" should be different from "Description added on field"

Reply:
> It could be overwhelming to see a ton of "Description added on new field" for every field on a type that was newly added, that might not be helpful.

The goal was to provide our docs team with a comprehensive set of all the new strings that needed reviewing; I'll type these differently when I add stronger types per your earlier comment, and then the caller can filter as desired.